tsdb/index.go

```diff
@@ -2512,6 +2509,18 @@ func (is IndexSet) tagValuesByKeyAndExpr(auth query.Authorizer, name []byte, key
 			break
 		}
 
+		if e.Expr != nil {
+			// We don't yet have code that correctly processes expressions that
+			// seriesByExprIterator doesn't handle
+			lit, ok := e.Expr.(*influxql.BooleanLiteral)
+			if !ok {
+				return nil, fmt.Errorf("Expression too complex for metaquery: %v", e.Expr)
```

Comment:
Wondering if there is a more informative error message we could give the user here. "Meta queries don't support..."?

Reply:
I'm not sure what the exact rules are unfortunately, and I'm worried this would be hard to maintain if we add more support in the various places we partially evaluate the query.